airflow/contrib/auth/backends/astronomer_auth.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+
+from sys import version_info
+
+import flask_login
+from flask_login import login_required, current_user, logout_user
+from flask import flash
+from wtforms import (
+    Form, PasswordField, StringField)
+from wtforms.validators import InputRequired
+
+from flask import url_for, redirect
+from flask_bcrypt import generate_password_hash, check_password_hash
+
+from sqlalchemy import (
+    Column, String, DateTime)
+from sqlalchemy.ext.hybrid import hybrid_property
+
+from airflow import settings
+from airflow import models
+from airflow import configuration
+
+import logging
+import requests
+import json
+
+login_manager = flask_login.LoginManager()
+login_manager.login_view = 'airflow.login'  # Calls login() below
+login_manager.login_message = None
+
+LOG = logging.getLogger(__name__)
+PY3 = version_info[0] == 3
+
+
+class AuthenticationError(Exception):
+    pass
+
+
+'''
+class User(Base):
+    __tablename__ = "users"
+
+    id = Column(Integer, primary_key=True)
+    username = Column(String(ID_LEN), unique=True)
+    email = Column(String(500))
+    superuser = False
+
+    def __repr__(self):
+        return self.username
+
+    def get_id(self):
+        return str(self.id)
+
+    def is_superuser(self):
+        return self.superuser
+'''
+
+class AstronomerUser(models.User):
+    def __init__(self, user):
+        self.user = user
+
+    @staticmethod
+    def authenticate(username, password):
+        # TODO: make API call to API server
+        # TODO: allow hostname/port/protocol to be configured, set sane default
+        hostname = configuration.get("astronomer-api", "hostname")
+        port = configuration.get("astronomer-api", "port")
+        protocol = configuration.get("astronomer-api", "protocol")
+        endpoint = '%s://%s:%s/v1' % (protocol, hostname, port)
```

Comment:
@perryao @cwurtz `str.format` is valid and preferred over `%` formatting.  I would do:

```python
endpoint = '{}://{}:{}/v1'.format(protocol, hostname, port)
```

I also like building the base URL separately from a particular endpoint for the sake of being DRY for future revisions, unit tests, etc.  So better yet:

```python
from requests.compat import urljoin

base = '{}://{}:{}'.format(protocol, hostname, port)
endpoint = urljoin(base, 'v1')
```

(If you import `requests.compat.urljoin`, it wraps `urllib` for compatibility with Python 2 and 3 vs importing from urllib directly.)